R/functions-xcmsSwath.R

```diff
@@ -0,0 +1,33 @@
+## function "align" EICs
+.alignEics <- function(x, y) {
+
+  # linear interpolation of y on x
+  y_aligned <- approx(rtime(y), intensity(y), rtime(x))
+  
+  # deal with NA
+  y_aligned[is.na()] <- 0
+  
+  # correct rtime and int in chromatogram object
+  y@rtime <- y_aligned$x
+  y@intensity <- y_aligned$y
+  
+  # return
+  return(y)
+  
+}
+
+# function to correlate EICs
+.correlateEics <- function(x, y) {
```

Comment:
Also here, IMHO it would be better to call that `.correlate_chromatogram` or `.correlateChromatogram`.

Reply:
I would also add here parameters of the `cor` function, i.e. `use = "pairwise.complete.obs"` and `method = c("pearson", "kendall", "spearman")`. These should then be passed along to the `cor` function.

Reply:
No problem.

Reply:
You want a `...` notation or explicit?

Reply:
I would add them as explicit parameters.